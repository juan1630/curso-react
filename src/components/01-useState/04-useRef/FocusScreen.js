import React, { useRef } from 'react'
import '../02-useEffect/useEffect.css';

export const FocusScreen = () => {

    const inputRef = useRef();
    // se puede modificar el ref

    const nandleClick = ()=> {
        // hace la referencia al input como el viewChild
        inputRef.current.select();  
        // hook use ref, es para referencia mutable
        console.log( inputRef );
        // se le puede dar el seguimiento a objetos, valores, etc.
        
    }
    return (
        <div>
            <h1> Focus Screen  </h1>
            <hr/>

            <input ref={   inputRef  } type="text"  className="form-control" placeholder="Nombre..." />

            <button  className="btn btn-primary mt-4" onClick={ nandleClick } >  Focus! </button>
        </div>
    )
}
