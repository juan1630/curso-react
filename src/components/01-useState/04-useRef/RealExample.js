import React, { useState } from 'react'
import '../02-useEffect/useEffect.css';
import {   MultiplaCustomHooks } from '../examples/MultiplaCustomHooks';


export const RealExample = () => {

   const [show, setShow] = useState(false);


    return (
        <div>
            <h1>  Real example ref  </h1>
            <hr/>

                { show &&  <MultiplaCustomHooks/> }

                <button  className="btn btn-primary"  onClick={ () => { 
                    setShow( !show ) 
                    
                    }}  >  Show/Hide </button>

        </div>
    )
}
