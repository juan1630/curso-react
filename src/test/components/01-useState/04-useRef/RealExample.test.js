import { shallow } from 'enzyme';
import React from 'react';
import { RealExample } from '../../../../components/01-useState/04-useRef/RealExample';


describe('pruebas en el componente realExample', () => {

    const wrapper = shallow( <RealExample /> )

    test('Debe de mostrar correctamente el componente', () => {
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultiplaCustomHooks').exists() ).toBe(false)
    })

    test('Debe de validar el componente <MultiplaCustomHooks/>', ()=> {

        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultiplaCustomHooks').exists() ).toBe(true);

    });

});