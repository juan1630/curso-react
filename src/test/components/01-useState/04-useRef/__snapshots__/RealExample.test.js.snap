// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente realExample Debe de mostrar correctamente el componente 1`] = `
<div>
  <h1>
      Real example ref  
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
      Show/Hide 
  </button>
</div>
`;
